src/com/google/enterprise/adaptor/database/DatabaseAdaptor.java

```diff
@@ -357,6 +364,8 @@ static void verifyColumnNames(Connection conn, String sqlConfig, String sql,
               "Matched column \"{0}\" as \"{1}\" in query {2}",
               new Object[] { match, actual, sqlConfig });
           targets.remove(match);
+          // TODO(bmj): What happens with MySQL here.
+          typeMap.put(match, rsmd.getColumnType(i));
```

Comment:
I had gotten as far as realizing that we could do the work with ResultSetMetaData rather than ParameterMetaData, but hadn't realized that we could avoid having to figure out which column the type was for this way.